feature_engine/transformation/arcsin.py

```diff
@@ -0,0 +1,178 @@
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._docstrings.class_inputs import _variables_numerical_docstring
+from feature_engine._docstrings.fit_attributes import (
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.methods import (
+    _fit_not_learn_docstring,
+    _fit_transform_docstring,
+    _inverse_transform_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.tags import _return_tags
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+@Substitution(
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+    inverse_transform=_inverse_transform_docstring,
+)
+class ArcsinTransformer(BaseNumericalTransformer):
+    """
+    The ArcsinTransformer() applies the arcsin transformation to numerical variables.
+
+    The ArcsinTransformer() only works with numerical variables between 0 and +1.
+    If a variable contains a value outside of this range, the transformer will
+    raise an error.
+
+    A list of variables can be passed as an argument. Alternatively, the transformer
+    will automatically select and transform all numerical variables.
+
+    More details in the :ref:`User Guide <arcsin>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    Attributes
+    ----------
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {inverse_transform}
+
+    transform:
+        Apply the arcsin transformation.
+
+    """
+
+    def __init__(
+        self, variables: Union[None, int, str, List[Union[str, int]]] = None
+    ) -> None:
+
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn parameters.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = super()._fit_from_varlist(X)
+
+        # check if the variables is in the correct range
+        if ((X[self.variables_] < 0) | (X[self.variables_] > 1)).any().any():
+            raise ValueError(
+                "Some variables contain values outside the possible range"
+                "can't apply arcsin transformation"
```

Comment:
Can we modify slightly the error message?

                "Some variables contain values outside the range 0-1. "
                "Can't apply arcsin transformation."

Important to leave the space after the . in the first line.